USERGUIDE.md

```diff
@@ -2435,13 +2443,143 @@ For VS Code users, the `"erase flash"` [build task](#build-tasks) can be used to
+- Decrease in setup overhead in a Transaction function call. Less nested function calls within drivers.
+- Improved SPI DMA support and DMA Channel IRQ vector flexibility by providing acquired DMA channel numbers before a SPI DMA transaction call.
+- The use of Controller and Target terms instead of Master and Slave, respectively.
+- Still supports SPI v1 function prototypes for backwards-compatibility.
+- Bug fixes from the SPI v1 API.
+
+#### Porting Projects to use SPI v2
+
+The latest SPI examples in the MSDK defaults to build the SPI v1 libraries. Set the `MXC_SPI_VERSION` [build configuration variable](#build-configuration-variables) to `v2` (case sensitive) use the SPI v2 API.
+
+This guide shows how to update an existing project that is using the SPI v1 API to SPI v2. The SPI v2 Library still supports the SPI v1 function prototypes for backwards-compatibility with the main difference in the SPI DMA interrupt handling (see [SPI DMA Interrupt Handling](#spi-dma-interrupt-handling) section below for more info).
+
+Note: The SPI v2 API is only a drop in replacement to SPI v1 if SPI DMA is **not** used; should the user choose to continue building with the SPI v1 convention but with the underlying SPI v2 implementation. This porting guide demonstrates how to use the full extent of the SPI v2 features.
+
+##### SPI Init Function
+
+The input parameters for the `MXC_SPI_Init(...)` function were updated in SPI v2 to allow for more user-selectable options during initialization. This should not cause any errors or behavioral differences with the `MXC_SPI_Init(...)` function when switching between SPI v1 and SPI v2 builds as the input parameters were essentially re-mapped to more descriptive names.
+
+SPI v1:
+
+    :::C
+    int MXC_SPI_Init(mxc_spi_regs_t *spi, 
+                    int masterMode, 
+                    int quadModeUsed, 
+                    int numSlaves, 
+                    unsigned ssPolarity, 
+                    unsigned int hz, 
+                    mxc_spi_pins_t pins)
+
+SPI v2:
+
+    :::C
+    int MXC_SPI_Init(mxc_spi_regs_t *spi, 
+                    mxc_spi_type_t controller_target, 
+                    mxc_spi_interface_t if_mode, 
+                    int numTargets, 
+                    uint8_t ts_active_pol_mask, 
+                    uint32_t freq, 
+                    mxc_spi_pins_t pins)
+
+Input Parameters:
+
+- `mxc_spi_regs_t *spi` remains unchanged.
+- `int masterMode` -> `mxc_spi_type_t controller_target`. The enum `mxc_spi_type_t` was added for increased code readability.
+- `int quadModeUsed` -> `mxc_spi_interface_t if_mode`. Previously, the `MXC_SPI_Init(...)` function could only select between standard (4wire) and quad interface modes. With SPI v2, the user can select either standard (`MXC_SPI_INTERFACE_STANDARD`), quad (`MXC_SPI_INTERFACE_QUAD`), 3wire (`MXC_SPI_INTERFACE_3WIRE`), or dual (`MXC_SPI_INTERFACE_DUAL`) mode.
```

Comment:
Consider the maintenance overhead for this document when you write out the actual values of the enum here.  I think it would be better to just write "this is now selected with an enum" to keep it generic.  Intellisense and/or the Doxygen will tell users what their options are.